src/libsyncengine/update_detection/file_system_observer/localfilesystemobserverworker.cpp

```diff
@@ -83,9 +83,16 @@ void LocalFileSystemObserverWorker::changesDetected(const std::list<std::pair<st
         _needUpdateTimerStart = std::chrono::steady_clock::now();
 
         OperationType opTypeFromOS = changedItem.second;
-        SyncPath absolutePath = changedItem.first.native();
-        SyncPath relativePath = CommonUtility::relativePath(_syncPal->localPath(), absolutePath);
+        const SyncPath absolutePath = changedItem.first.native();
+        const SyncPath relativePath = CommonUtility::relativePath(_syncPal->localPath(), absolutePath);
 
+        if (_syncPal->isTmpBlacklisted(relativePath, ReplicaSide::Local)) {
```

Comment:
The nodeId is extracted a bit later.
It should be better to check isTmpBlacklisted by nodeId (key of the map).

Reply:
Done here: [e637b94](https://github.com/Infomaniak/desktop-kDrive/pull/356/commits/e637b94214070ffd08e28dd9945ad086c069a4bf)

Reply:
Sorry to re-open but did you apply the comment? We are still using the path here.
Also, `isTmpBlacklisted` is called again in case of Delete AND again in other cases. Is this call really necessary?

Reply:
Thank you for spotting it, I should have removed it frome here!
Done here: [64c01a5](https://github.com/Infomaniak/desktop-kDrive/pull/356/commits/64c01a5461790b485612afcf17d8b2e15831f6e5)

Reply:
Perfect thx